lib/ansible/modules/windows/win_power_plan.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2015, Jon Hawkesworth (@jhawkesworth) <figs@unity.demon.co.uk>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = r'''
+---
+module: win_power_plan
+author: "Noah Sparks (@nwsparks)"
+version_added: "2.3.2"
+short_description: Changes the power plan of a Windows system.
+description:
+- win_power_plan will change the power plan of a Windows system to the defined string.
+- Windows defaults to "balanced" which will cause CPU throttling. In some cases it can be preferable
+  to change the mode to "high performance" to increase CPU performance
+options:
+  power_plan:
+    description:
+    - String value that indicates the desired power plan. The power plan must already be
+      present on the system. Commonly there will be options for "balanced" and "high performance"
+    - Defaults to "high performance"
+    - Returns all available power plans to stdout in results
+    version_added: "2.3"
+'''
+
+EXAMPLES = r'''
+- name: change power plan to high performance
+  win_power_plan:
+    power_plan: high performance
+'''
+
+RETURN = r'''
+power_plan:
+    description: Value of the intended power plan
+    returned: always
+    type: string
+    sample: balanced
+power_plan_enabled:
+    description: State of the intended power plan
+    returned: success
+    type: boolean
+    sample: True
+all_available_plans:
+    description: source file used for the copy on the target machine
+    returned: always
+    type: array
+    sample: |
+      "all_available_plans": [
```

Comment:
Thanks for the reviews. I'll see if I can convert it. Was just converting the powershell object to json and this was the result, should be able to convert it to a dictionary first though. Will fix documentation errors as well. Also updating the integration test a bit. Will commit again once ready.